packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -646,6 +646,9 @@ abstract class ResidentRunner {
       final bool handled = await _commonTerminalInputHandler(command);
       if (!handled)
         await handleTerminalCommand(command);
+    } catch (error) {
+      _resetTerminal();
+      rethrow;
```

Comment:
Who is going to catch this exception?
If we reset the terminal shouldn't we remove the listener too?